### PR TITLE
Draft: Add an algorithm for scfi merger

### PR DESCRIPTION
**Pull request by Chao1009**
### Briefly, what does this PR introduce?
To resolve issue #465 

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #465)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No, this PR introduced a new merger algorithm for the ScFi Barrel Calorimeter, which was well communicated with the current BECal users.

### Does this PR change default behavior?
It supposes to replace the current merger in the reconstruction steering file for the ScFi Barrel Calorimeter.

### CODE REVIEW

**Comment by Chao1009** [comment]
@mariakzurek

**Comment by mariakzurek** [comment]
Thanks, @Chao1009. I see that you are adding a new algorithm that is merging info from fibers and applies the digitization parameters. Is there a reason to do that, instead of using existing merging and digi parameter? (Is it about the CellID)?

File: src/algorithms/calorimetry/CalorimeterScFiDigi.cc [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    // Delete any output objects left from last event.
+    // (Should already have been done for us, but just to be bullet-proof.)
+    for( auto h : rawhits ) delete h;
+    rawhits.clear();
+
+    light_guide_digi();
+}
+
+//------------------------
+// signal_sum_digi
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterScFiDigi.cc:132]: Fix name.

Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterScFiDigi.cc:132]:
```suggestion
// light_guide_digi
```

**Review by wdconinc — APPROVED** [review]

**Comment by wdconinc** [comment]
Based on discussion with @mariakzurek, this is ready for merging.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**